src/main/java/com/blamejared/crafttweaker/api/recipes/GatherReplacementExclusionEvent.java

```diff
@@ -0,0 +1,76 @@
+package com.blamejared.crafttweaker.api.recipes;
+
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.eventbus.api.Event;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Fired whenever replacement exclusions for a specific {@link IRecipeManager} need to be gathered.
+ *
+ * <p>Recipe names that are added in this event will then be excluded from the set of recipes that {@code Replacer}s are
+ * allowed to modify, meaning that users will have to remove and add recipes manually instead of being able to resort to
+ * automated tools.</p>
+ *
+ * <p>Mods should add recipes to this event <strong>if and only if</strong> replacement is actually impossible, and not
+ * based on whether replacement is supported on their end or not. As an example, {@code minecraft:armor_dye} is a recipe
+ * that is hardcoded and thus cannot have its ingredients replaced: this is a good candidate to exclude.</p>
```

Comment:
I think it would be helpful to have this also mention the fact that CrT automatically adds special recipes to being excluded so that people don't think they need to include their regular crafting table special recipes.